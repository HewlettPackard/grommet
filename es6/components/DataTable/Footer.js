var _excluded = ["cellProps", "columns", "fill", "footerValues", "groups", "onSelect", "pin", "pinnedOffset", "primaryProperty", "selected", "verticalAlign"];
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (e.indexOf(n) >= 0) continue; t[n] = r[n]; } return t; }
import React, { forwardRef } from 'react';
import { TableRow } from '../TableRow';
import { TableCell } from '../TableCell';
import { Cell } from './Cell';
import { StyledDataTableCell, StyledDataTableFooter } from './StyledDataTable';
var Footer = /*#__PURE__*/forwardRef(function (_ref, ref) {
  var cellProps = _ref.cellProps,
    columns = _ref.columns,
    fill = _ref.fill,
    footerValues = _ref.footerValues,
    groups = _ref.groups,
    onSelect = _ref.onSelect,
    pinProp = _ref.pin,
    pinnedOffset = _ref.pinnedOffset,
    primaryProperty = _ref.primaryProperty,
    selected = _ref.selected,
    verticalAlign = _ref.verticalAlign,
    rest = _objectWithoutPropertiesLoose(_ref, _excluded);
  var pin = pinProp ? ['bottom'] : [];
  return /*#__PURE__*/React.createElement(StyledDataTableFooter, _extends({
    ref: ref,
    fillProp: fill,
    pin: pinProp
  }, rest), /*#__PURE__*/React.createElement(TableRow, null, groups && /*#__PURE__*/React.createElement(TableCell, {
    plain: true,
    size: "xxsmall",
    pad: "none",
    verticalAlign: "top",
    background: cellProps.background,
    border: cellProps.border
  }), (selected || onSelect) && /*#__PURE__*/React.createElement(StyledDataTableCell, {
    background: cellProps.background,
    context: "footer",
    pin: pin,
    verticalAlign: verticalAlign
  }), columns.map(function (column) {
    var cellPin = [].concat(pin);
    if (column.pin) cellPin.push('left');
    return /*#__PURE__*/React.createElement(Cell, {
      key: column.property,
      background: column.pin && cellProps.pinned.background || cellProps.background,
      border: column.pin && cellProps.pinned.border || cellProps.border,
      context: "footer",
      column: column,
      datum: footerValues,
      pad: column.pin && cellProps.pinned.pad || cellProps.pad,
      pin: pin.length ? pin : undefined,
      pinnedOffset: pinnedOffset && pinnedOffset[column.property],
      primaryProperty: primaryProperty,
      verticalAlign: verticalAlign
    });
  })));
});
Footer.displayName = 'Footer';
export { Footer };