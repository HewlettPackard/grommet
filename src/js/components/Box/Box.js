import React, { Children, Component } from 'react';
import PropTypes from 'prop-types';
import { compose } from 'recompose';

import { colorForName, colorIsDark } from '../../utils';

import { withTheme } from '../hocs';

import StyledBox, { StyledBoxGap } from './StyledBox';

import doc from './doc';

const styledComponents = {
  div: StyledBox,
}; // tag -> styled component

class Box extends Component {
  static contextTypes = {
    grommet: PropTypes.object,
  }
  static childContextTypes = {
    grommet: PropTypes.object,
  }

  static defaultProps = {
    direction: 'column',
    margin: 'none',
    pad: 'none',
    tag: 'div',
  };

  getChildContext() {
    const { grommet } = this.context;
    const { background, theme } = this.props;
    let dark = false;
    if (background) {
      if (typeof background === 'object') {
        dark = background.dark;
      } else {
        const color = colorForName(background, theme);
        if (color) {
          dark = colorIsDark(color);
        }
      }
      return {
        grommet: { ...grommet, dark },
      };
    }
    return {};
  }

  render() {
    const {
      a11yTitle,
      children,
      direction,
      gap,
      tag,
      theme,
      ...rest
    } = this.props;

    let StyledComponent = styledComponents[tag];
    if (!StyledComponent) {
      StyledComponent = StyledBox.withComponent(tag);
      styledComponents[tag] = StyledComponent;
    }

    let contents = children;
    if (gap) {
      contents = [];
      let firstIndex;
      Children.forEach(children, (child, index) => {
        if (child) {
          if (firstIndex === undefined) {
            firstIndex = index;
          } else {
            contents.push((
              <StyledBoxGap
                key={index}
                gap={gap}
                direction={direction}
                theme={theme}
              />
            ));
          }
        }
        contents.push(child);
      });
    }

    return (
      <StyledComponent
        aria-label={a11yTitle}
        direction={direction}
        theme={theme}
        {...rest}
      >
        {contents}
      </StyledComponent>
    );
  }
}

if (process.env.NODE_ENV !== 'production') {
  doc(Box);
}

export default compose(
  withTheme,
)(Box);
