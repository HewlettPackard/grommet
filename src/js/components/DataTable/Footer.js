import React from 'react';

import { defaultProps } from '../../default-props';

import { TableRow } from '../TableRow';
import { TableCell } from '../TableCell';

import { Cell } from './Cell';
import { StyledDataTableFooter } from './StyledDataTable';

const Footer = ({
  background,
  border,
  columns,
  fill,
  footerValues,
  groups,
  pad,
  pin: tablePin,
  primaryProperty,
  ...rest
}) => (
  <StyledDataTableFooter fillProp={fill} {...rest}>
    <TableRow>
      {groups && (
        <TableCell plain size="xxsmall" pad="none" verticalAlign="top" />
      )}
      {columns.map(column => {
        const pin = [];
        if (tablePin) pin.push('bottom');
        if (column.pin) pin.push('left');
        return (
          <Cell
            key={column.property}
            background={background}
            border={border}
            context="footer"
            column={column}
            datum={footerValues}
            pad={pad}
            pin={pin}
            primaryProperty={primaryProperty}
          />
        );
      })}
    </TableRow>
  </StyledDataTableFooter>
);

Footer.displayName = 'Footer';

Footer.defaultProps = {};
Object.setPrototypeOf(Footer.defaultProps, defaultProps);

export { Footer };
