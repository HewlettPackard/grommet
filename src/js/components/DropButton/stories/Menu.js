import React from 'react';
import { storiesOf } from '@storybook/react';

import { MnetUIBase, Box, DropButton, Text } from 'mnet-ui-base';
import { Gremlin } from 'grommet-icons';
import { mnet } from 'mnet-ui-base/themes';

const renderItems = () => (
  <Box>
    <Text>hi</Text>
    <Text>hi</Text>
    <Text>hi</Text>
    <Text>hi</Text>
  </Box>
);

const MenuItem = () => (
  <Box height="36px" width="36px" align="center">
    <Gremlin />
  </Box>
);

const GremlinDropButton = () => (
  <DropButton
    alignSelf="center"
    margin={{ vertical: 'small' }}
    dropContent={renderItems()}
    dropProps={{ align: { top: 'bottom' } }}
  >
    <MenuItem />
  </DropButton>
);
const MenuDropButton = () => {
  return (
    <MnetUIBase theme={mnet} full>
      <Box fill>
        <Box fill="vertical" width="xxsmall" background="dark-2">
          <GremlinDropButton />
          <Box flex />
          <GremlinDropButton />
        </Box>
      </Box>
    </MnetUIBase>
  );
};

storiesOf('DropButton', module).add('Menu', () => <MenuDropButton />);
