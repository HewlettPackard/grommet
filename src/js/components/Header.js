// (C) Copyright 2014-2015 Hewlett-Packard Development Company, L.P.

var React = require('react');
var merge = require('lodash/object/merge');
var pick = require('lodash/object/pick');
var keys = require('lodash/object/keys');
var Box = require('./Box');

var CLASS_ROOT = "header";

var Header = React.createClass({

  propTypes: merge({
    fixed: React.PropTypes.bool,
    float: React.PropTypes.bool,
    fullWidth: React.PropTypes.bool,
    large: React.PropTypes.bool,
    size: React.PropTypes.oneOf(['small', 'medium', 'large']),
    small: React.PropTypes.bool,
    splash: React.PropTypes.bool,
    strong: React.PropTypes.bool,
    tag: React.PropTypes.string
  }, Box.propTypes),

  getDefaultProps: function () {
    return {
      pad: 'none',
      direction: 'row',
      fullWidth: true,
      align: 'center',
      responsive: false,
      tag: 'header'
    };
  },

  getInitialState: function() {
    return this._stateFromProps(this.props);
  },

  componentDidMount: function () {
    if (this.props.fixed) {
      this._alignMirror();
      window.addEventListener('resize', this._onResize);
    }
  },

  componentWillReceiveProps: function (newProps) {
    this.setState(this._stateFromProps(newProps));
  },

  componentDidUpdate: function () {
    if (this.props.fixed) {
      this._alignMirror();
    }
  },

  componentWillUnmount: function () {
    if (this.props.fixed) {
      window.removeEventListener('resize', this._onResize);
    }
  },

  _stateFromProps: function (props) {
    return {size: props.size || (props.small ? 'small' : (props.large ? 'large' : null))};
  },

  _onResize: function () {
    this._alignMirror();
  },

  _alignMirror: function () {
    var contentElement = this.refs.content.getDOMNode();
    var mirrorElement = this.refs.mirror.getDOMNode();

    // constrain fixed content to the width of the mirror
    var mirrorRect = mirrorElement.getBoundingClientRect();
    contentElement.style.width = '' + Math.floor(mirrorRect.width) + 'px';

    // align the mirror height with the content's height
    var contentRect = contentElement.getBoundingClientRect();
    mirrorElement.style.height = '' + Math.floor(contentRect.height) + 'px';
  },

  render: function() {
    var classes = [CLASS_ROOT];
    var containerClasses = [CLASS_ROOT + "__container"];
    var other = pick(this.props, keys(Box.propTypes));
    if (this.props.fixed) {
      containerClasses.push(CLASS_ROOT + "__container--fixed");
    }
    if (this.props.float) {
      classes.push(CLASS_ROOT + "--float");
      containerClasses.push(CLASS_ROOT + "__container--float");
    }
    if (this.props.fullWidth) {
       classes.push(CLASS_ROOT + "--full-width");
     }
    if (this.state.size) {
      classes.push(CLASS_ROOT + "--" + this.state.size);
    }
    if (this.props.splash) {
      classes.push(CLASS_ROOT + "--splash");
    }
    if (this.props.strong) {
      classes.push(CLASS_ROOT + "--strong");
    }
    if (this.props.className) {
      classes.push(this.props.className);
    }

    if (this.props.fixed) {
      return (
        <div className={containerClasses.join(' ')}>
          <div ref="mirror" className={CLASS_ROOT + "__mirror"}></div>
          <div className={CLASS_ROOT + "__wrapper"}>
            <Box ref="content" tag={this.props.header} {...other} className={classes.join(' ')}>
              {this.props.children}
            </Box>
          </div>
        </div>
      );
    } else {
      return (
        <Box tag={this.props.header} {...other} className={classes.join(' ')}
          containerClassName={containerClasses.join(' ')}>
          {this.props.children}
        </Box>
      );
    }
  }

});

module.exports = Header;
