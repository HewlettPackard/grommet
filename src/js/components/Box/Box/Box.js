import React from 'react';
import PropTypes from 'prop-types';
import { Box as BoxM } from '../Box';

function Box(props) {
  return <BoxM {...props} />
}

/* PropTypes for UXPin Merge */
Box.propTypes = {
  children: PropTypes.node,
  a11yTitle: PropTypes.string,
  alignSelf: PropTypes.oneOf(["start", "center", "end", "stretch"]),
  gridArea: PropTypes.string,
  margin: PropTypes.oneOf(["none", "xxsmall", "xsmall", "small", "medium", "large", "xlarge"]),
  align: PropTypes.oneOf(["start", "center", "end", "baseline", "stretch"]),
  alignContent: PropTypes.oneOf(["start", "center", "end", "between", "around", "stretch"]),
  animation: PropTypes.oneOf(["fadeIn", "fadeOut", "jiggle", "pulse", "slideUp", "slideDown", "slideLeft", "slideRight", "zoomIn", "zoomOut"]),
  background: PropTypes.string,
  basis: PropTypes.oneOf(["xxsmall", "xsmall", "small", "medium", "large", "xlarge", "xxlarge", "full", "1/2", "1/3", "2/3", "1/4", "2/4", "3/4", "auto"]),
  border: PropTypes.oneOf([ "top", "left", "bottom", "right", "horizontal", "vertical",  "all"]),
  direction: PropTypes.oneOf(["row", "column", "row-responsive"]),
  elevation: PropTypes.oneOf(["none", "xsmall", "small", "medium", "large", "xlarge"]),
  flex: PropTypes.oneOf(["grow", "shrink"]),
  fill: PropTypes.oneOf(["horizontal", "vertical"]),
  gap: PropTypes.oneOf(["xxsmall", "xsmall", "small", "medium", "large", "xlarge"]),
  height: PropTypes.oneOf(["xxsmall", "xsmall", "small", "medium", "large", "xlarge", "xxlarge"]),
  justify: PropTypes.oneOf(["start", "center", "between", "around", "evenly", "end"]),
  overflow: PropTypes.oneOf(["auto", "hidden", "scroll", "visible"]),
  pad: PropTypes.oneOf(["none", "xxsmall", "xsmall", "small", "medium", "large", "xlarge"]),
  responsive: PropTypes.bool,
  round: PropTypes.bool,
  tag: PropTypes.string,
  as: PropTypes.string,
  width: PropTypes.oneOf(["xxsmall", "xsmall", "small", "medium", "large", "xlarge", "xxlarge"]),
  wrap: PropTypes.bool,
}

/* Export Default for UXPin Merge  */
export default Box;