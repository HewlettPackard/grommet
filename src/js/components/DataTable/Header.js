import React, { useContext } from 'react';
import { ThemeContext } from 'styled-components';

import { defaultProps } from '../../default-props';

import { Box } from '../Box';
import { Button } from '../Button';
import { Text } from '../Text';

import { Resizer } from './Resizer';
import { Searcher } from './Searcher';
import { ExpanderCell } from './ExpanderCell';
import {
  StyledDataTableCell,
  StyledDataTableHeader,
  StyledDataTableRow,
} from './StyledDataTable';

const Header = ({
  background,
  border,
  columns,
  fill,
  filtering,
  filters,
  groups,
  groupState,
  onFilter,
  onFiltering,
  onResize,
  onSort,
  onToggle,
  pad,
  pin: tablePin,
  sort,
  widths,
  ...rest
}) => {
  const theme = useContext(ThemeContext) || defaultProps.theme;
  return (
    <StyledDataTableHeader fillProp={fill} {...rest}>
      <StyledDataTableRow>
        {groups && (
          <ExpanderCell
            context="header"
            expanded={
              Object.keys(groupState).filter(k => !groupState[k].expanded)
                .length === 0
            }
            onToggle={onToggle}
          />
        )}

        {columns.map(
          ({
            property,
            header,
            align,
            pin: columnPin,
            search,
            sortable,
            verticalAlign,
            size,
          }) => {
            let content =
              typeof header === 'string' ? <Text>{header}</Text> : header;

            if (onSort && sortable !== false) {
              const Icon =
                onSort &&
                sortable !== false &&
                sort &&
                sort.property === property &&
                theme.dataTable.icons[
                  sort.direction !== 'asc' ? 'ascending' : 'descending'
                ];
              content = (
                <Button plain fill="vertical" onClick={onSort(property)}>
                  <Box direction="row" align="center" gap="xsmall">
                    {content}
                    {Icon && <Icon />}
                  </Box>
                </Button>
              );
            }

            if (search || onResize) {
              const resizer = onResize ? (
                <Resizer property={property} onResize={onResize} />
              ) : null;
              const searcher =
                search && filters ? (
                  <Searcher
                    filtering={filtering}
                    filters={filters}
                    property={property}
                    onFilter={onFilter}
                    onFiltering={onFiltering}
                  />
                ) : null;
              content = (
                <Box
                  direction="row"
                  align="center"
                  justify={!align || align === 'start' ? 'between' : align}
                  gap="small"
                  fill="vertical"
                  style={onResize ? { position: 'relative' } : undefined}
                >
                  {content}
                  {searcher && resizer ? (
                    <Box
                      flex="shrink"
                      direction="row"
                      align="center"
                      gap="small"
                    >
                      {searcher}
                      {resizer}
                    </Box>
                  ) : (
                    searcher || resizer
                  )}
                </Box>
              );
            }
            const pin = [];
            if (tablePin) pin.push('top');
            if (columnPin) pin.push('left');
            return (
              <StyledDataTableCell
                key={property}
                align={align}
                verticalAlign={verticalAlign}
                background={background}
                border={border}
                pad={pad}
                pin={pin}
                plain
                scope="col"
                size={widths && widths[property] ? undefined : size}
                style={
                  widths && widths[property]
                    ? { width: widths[property] }
                    : undefined
                }
              >
                {content}
              </StyledDataTableCell>
            );
          },
        )}
      </StyledDataTableRow>
    </StyledDataTableHeader>
  );
};

Header.displayName = 'Header';

Header.defaultProps = {};
Object.setPrototypeOf(Header.defaultProps, defaultProps);

export { Header };
