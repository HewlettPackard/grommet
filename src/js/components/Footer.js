// (C) Copyright 2014-2015 Hewlett Packard Enterprise Development LP

import React, { PropTypes } from 'react';
import classnames from 'classnames';
import Box from './Box';
import SkipLinkAnchor from './SkipLinkAnchor';
import Props from '../utils/Props';

const CLASS_ROOT = 'footer';

const Footer = props => {

  let classes = classnames(
    CLASS_ROOT,
    props.className,
    {
      [`${CLASS_ROOT}--${props.size}`]: props.size,
      [`${CLASS_ROOT}--float`]: props.float
    }
  );

  let containerClasses = classnames(
    `${CLASS_ROOT}__container`,
    {
      [`${CLASS_ROOT}__container--float`]: props.float
    }
  );

  let footerSkipLink;
  if (props.primary) {
    footerSkipLink = <SkipLinkAnchor label="Footer" />;
  }

  let boxProps = Props.pick(props, Box);

  return (
    <Box {...boxProps} tag="footer" className={classes}
      containerClassName={containerClasses}
      primary={false}>
      {footerSkipLink}
      {props.children}
    </Box>
  );
};

Footer.propTypes = {
  float: PropTypes.bool,
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  primary: PropTypes.bool,
  ...Box.propTypes
};

Footer.defaultProps = {
  direction: 'row',
  responsive: false
};

Footer.displayName = 'Footer';

export default Footer;
