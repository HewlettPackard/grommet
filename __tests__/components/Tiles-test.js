// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react/lib/ReactTestRenderer';

import Tiles from '../../src/js/components/Tiles';
import Tile from '../../src/js/components/Tile';
import Box from '../../src/js/components/Box';
import Header from '../../src/js/components/Header';
import Footer from '../../src/js/components/Footer';
import Menu from '../../src/js/components/Menu';

// needed because this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('Tiles', () => {
  it('has correct default options', () => {
    const component = renderer.create(
      <Tiles>
        <Tile>
          First
        </Tile>
        <Tile>
          Second
        </Tile>
        <Tile>
          Third
        </Tile>
      </Tiles>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders with the Not flush, Centered Example', () => {
    const component = renderer.create(
      <Tiles flush={false}
        justify="center" colorIndex="light-2" full="horizontal">
        <Tile align="start" colorIndex="light-1">
          <Header tag="h4" size="small" pad={{"horizontal": "small"}}>
            <strong>
              Tile 1
            </strong>
          </Header>
          <Box pad="small">
            <p>
              Tile summary content.
            </p>
          </Box>
          <Footer justify="between">
            <span />
            <Menu inline={false}
              dropAlign={{"bottom": "bottom", "right": "right"}}>
              <a>
                action 1
              </a>
              <a>
                action 2
              </a>
              <a>
                action 3
              </a>
            </Menu>
          </Footer>
        </Tile>
        <Tile align="start" colorIndex="light-1">
          <Header tag="h4" size="small" pad={{"horizontal": "small"}}>
            <strong>
              Tile 2
            </strong>
          </Header>
          <Box pad="small">
            <p>
              Tile summary content.
            </p>
          </Box>
          <Footer justify="between">
            <span />
            <Menu inline={false}
              dropAlign={{"bottom": "bottom", "right": "right"}}>
              <a>
                action 1
              </a>
              <a>
                action 2
              </a>
              <a>
                action 3
              </a>
            </Menu>
          </Footer>
        </Tile>
      </Tiles>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders with the Fill, Flush Example.', () => {
    const component = renderer.create(
      <Tiles fill={true}>
        <Tile align="start" colorIndex="neutral-1">
          <Header tag="h4" size="small" pad={{"horizontal": "small"}}>
            <strong>
              Tile 1
            </strong>
          </Header>
          <Box pad="small">
            <p>
              Tile summary content.
            </p>
          </Box>
        </Tile>
        <Tile align="start" colorIndex="neutral-2">
          <Header tag="h4" size="small" pad={{"horizontal": "small"}}>
            <strong>
              Tile 2
            </strong>
          </Header>
          <Box pad="small">
            <p>
              Tile summary content.
            </p>
          </Box>
        </Tile>
        <Tile align="start" colorIndex="neutral-3">
          <Header tag="h4" size="small" pad={{"horizontal": "small"}}>
            <strong>
              Tile 3
            </strong>
          </Header>
          <Box pad="small">
            <p>
              Tile summary content.
            </p>
          </Box>
        </Tile>
      </Tiles>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
