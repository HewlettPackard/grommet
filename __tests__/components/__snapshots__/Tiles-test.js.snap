exports[`Tiles has correct default options 1`] = `
<div
  className="grommetux-box grommetux-box--direction-row grommetux-box--justify-start grommetux-box--responsive grommetux-box--pad-small grommetux-box--wrap grommetux-tiles grommetux-tiles--flush"
  id={undefined}
  onClick={undefined}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}>
  <div
    className="grommetux-box grommetux-box--direction-column grommetux-box--align-center grommetux-box--responsive grommetux-box--pad-none grommetux-tile grommetux-none-hover-color-index-disabled grommetux-tile--hover-border-small"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}>
    First
  </div>
  <div
    className="grommetux-box grommetux-box--direction-column grommetux-box--align-center grommetux-box--responsive grommetux-box--pad-none grommetux-tile grommetux-none-hover-color-index-disabled grommetux-tile--hover-border-small"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}>
    Second
  </div>
  <div
    className="grommetux-box grommetux-box--direction-column grommetux-box--align-center grommetux-box--responsive grommetux-box--pad-none grommetux-tile grommetux-none-hover-color-index-disabled grommetux-tile--hover-border-small"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}>
    Third
  </div>
</div>
`;

exports[`Tiles renders with the Fill, Flush Example. 1`] = `
<div
  className="grommetux-box grommetux-box--direction-row grommetux-box--justify-start grommetux-box--responsive grommetux-box--pad-small grommetux-box--wrap grommetux-tiles grommetux-tiles--fill grommetux-tiles--flush"
  id={undefined}
  onClick={undefined}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}>
  <div
    className="grommetux-box grommetux-box--direction-column grommetux-box--align-start grommetux-box--responsive grommetux-box--pad-none grommetux-background-color-index-neutral-1 grommetux-tile grommetux-none-hover-color-index-disabled grommetux-tile--hover-border-small"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}>
    <h4
      className="grommetux-box grommetux-box--direction-row grommetux-box--align-center grommetux-box--pad-horizontal-small grommetux-header grommetux-header--small"
      id={undefined}
      onClick={undefined}
      role={undefined}
      style={Object {}}
      tabIndex={undefined}>
      <strong>
        Tile 1
      </strong>
    </h4>
    <div
      className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-small"
      id={undefined}
      onClick={undefined}
      role={undefined}
      style={Object {}}
      tabIndex={undefined}>
      <p>
        Tile summary content.
      </p>
    </div>
  </div>
  <div
    className="grommetux-box grommetux-box--direction-column grommetux-box--align-start grommetux-box--responsive grommetux-box--pad-none grommetux-background-color-index-neutral-2 grommetux-tile grommetux-none-hover-color-index-disabled grommetux-tile--hover-border-small"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}>
    <h4
      className="grommetux-box grommetux-box--direction-row grommetux-box--align-center grommetux-box--pad-horizontal-small grommetux-header grommetux-header--small"
      id={undefined}
      onClick={undefined}
      role={undefined}
      style={Object {}}
      tabIndex={undefined}>
      <strong>
        Tile 2
      </strong>
    </h4>
    <div
      className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-small"
      id={undefined}
      onClick={undefined}
      role={undefined}
      style={Object {}}
      tabIndex={undefined}>
      <p>
        Tile summary content.
      </p>
    </div>
  </div>
  <div
    className="grommetux-box grommetux-box--direction-column grommetux-box--align-start grommetux-box--responsive grommetux-box--pad-none grommetux-background-color-index-neutral-3 grommetux-tile grommetux-none-hover-color-index-disabled grommetux-tile--hover-border-small"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}>
    <h4
      className="grommetux-box grommetux-box--direction-row grommetux-box--align-center grommetux-box--pad-horizontal-small grommetux-header grommetux-header--small"
      id={undefined}
      onClick={undefined}
      role={undefined}
      style={Object {}}
      tabIndex={undefined}>
      <strong>
        Tile 3
      </strong>
    </h4>
    <div
      className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-small"
      id={undefined}
      onClick={undefined}
      role={undefined}
      style={Object {}}
      tabIndex={undefined}>
      <p>
        Tile summary content.
      </p>
    </div>
  </div>
</div>
`;

exports[`Tiles renders with the Not flush, Centered Example 1`] = `
<div
  className="grommetux-box grommetux-box--full-horizontal grommetux-box--direction-row grommetux-box--justify-center grommetux-box--responsive grommetux-box--pad-small grommetux-box--wrap grommetux-background-color-index-light-2 grommetux-tiles"
  id={undefined}
  onClick={undefined}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}>
  <div
    className="grommetux-box grommetux-box--direction-column grommetux-box--align-start grommetux-box--responsive grommetux-box--pad-none grommetux-background-color-index-light-1 grommetux-tile grommetux-none-hover-color-index-disabled grommetux-tile--hover-border-large"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}>
    <h4
      className="grommetux-box grommetux-box--direction-row grommetux-box--align-center grommetux-box--pad-horizontal-small grommetux-header grommetux-header--small"
      id={undefined}
      onClick={undefined}
      role={undefined}
      style={Object {}}
      tabIndex={undefined}>
      <strong>
        Tile 1
      </strong>
    </h4>
    <div
      className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-small"
      id={undefined}
      onClick={undefined}
      role={undefined}
      style={Object {}}
      tabIndex={undefined}>
      <p>
        Tile summary content.
      </p>
    </div>
    <footer
      className="grommetux-box grommetux-box--direction-row grommetux-box--justify-between grommetux-box--pad-none grommetux-footer"
      id={undefined}
      onClick={undefined}
      role={undefined}
      style={Object {}}
      tabIndex={undefined}>
      <span />
      <div>
        <button
          aria-label="Open  Menu"
          className="grommetux-button grommetux-menu grommetux-menu--column grommetux-menu--controlled grommetux-menu__control grommetux-button--plain grommetux-button--icon"
          disabled={false}
          href={undefined}
          id={undefined}
          onClick={[Function bound _onOpen]}
          type="button">
          <span
            className="grommetux-button__icon">
            <svg
              aria-labelledby="more-title"
              className="grommetux-control-icon grommetux-control-icon-more"
              height="24px"
              role="img"
              version="1.1"
              viewBox="0 0 24 24"
              width="24px">
              <title
                id="more-title">
                <span
                  id="more">
                  more
                </span>
              </title>
              <g>
                <rect
                  fill="none"
                  height="24"
                  width="24"
                  x="0"
                  y="0" />
                <rect
                  height="4"
                  width="4"
                  x="0"
                  y="10" />
                <rect
                  height="4"
                  width="4"
                  x="10"
                  y="10" />
                <rect
                  height="4"
                  width="4"
                  x="20"
                  y="10" />
              </g>
            </svg>
          </span>
        </button>
      </div>
    </footer>
  </div>
  <div
    className="grommetux-box grommetux-box--direction-column grommetux-box--align-start grommetux-box--responsive grommetux-box--pad-none grommetux-background-color-index-light-1 grommetux-tile grommetux-none-hover-color-index-disabled grommetux-tile--hover-border-large"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}>
    <h4
      className="grommetux-box grommetux-box--direction-row grommetux-box--align-center grommetux-box--pad-horizontal-small grommetux-header grommetux-header--small"
      id={undefined}
      onClick={undefined}
      role={undefined}
      style={Object {}}
      tabIndex={undefined}>
      <strong>
        Tile 2
      </strong>
    </h4>
    <div
      className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-small"
      id={undefined}
      onClick={undefined}
      role={undefined}
      style={Object {}}
      tabIndex={undefined}>
      <p>
        Tile summary content.
      </p>
    </div>
    <footer
      className="grommetux-box grommetux-box--direction-row grommetux-box--justify-between grommetux-box--pad-none grommetux-footer"
      id={undefined}
      onClick={undefined}
      role={undefined}
      style={Object {}}
      tabIndex={undefined}>
      <span />
      <div>
        <button
          aria-label="Open  Menu"
          className="grommetux-button grommetux-menu grommetux-menu--column grommetux-menu--controlled grommetux-menu__control grommetux-button--plain grommetux-button--icon"
          disabled={false}
          href={undefined}
          id={undefined}
          onClick={[Function bound _onOpen]}
          type="button">
          <span
            className="grommetux-button__icon">
            <svg
              aria-labelledby="more-title"
              className="grommetux-control-icon grommetux-control-icon-more"
              height="24px"
              role="img"
              version="1.1"
              viewBox="0 0 24 24"
              width="24px">
              <title
                id="more-title">
                <span
                  id="more">
                  more
                </span>
              </title>
              <g>
                <rect
                  fill="none"
                  height="24"
                  width="24"
                  x="0"
                  y="0" />
                <rect
                  height="4"
                  width="4"
                  x="0"
                  y="10" />
                <rect
                  height="4"
                  width="4"
                  x="10"
                  y="10" />
                <rect
                  height="4"
                  width="4"
                  x="20"
                  y="10" />
              </g>
            </svg>
          </span>
        </button>
      </div>
    </footer>
  </div>
</div>
`;
